api/src/DataPersister/Util/DataPersisterObservable.php

```diff
@@ -0,0 +1,151 @@
+<?php
+
+namespace App\DataPersister\Util;
+
+use ApiPlatform\Core\DataPersister\ContextAwareDataPersisterInterface;
+use Closure;
+use Symfony\Component\HttpFoundation\RequestStack;
+
+class DataPersisterObservable {
+    private Closure $onBeforeCreate;
+    private Closure $onAfterCreate;
+    private Closure $onBeforeUpdate;
+    private Closure $onAfterUpdate;
+    private Closure $onBeforeRemove;
+    private Closure $onAfterRemove;
+    /**
+     * @var CustomActionListener[]
+     */
+    private array $customActionListeners = [];
+    /**
+     * @var PropertyChangeListener[]
+     */
+    private array $propertyChangedListeners = [];
+
+    public function __construct(
+        private ContextAwareDataPersisterInterface $dataPersister,
+        private RequestStack $requestStack
+    ) {
+        $identity = fn ($data) => $data;
+        $noop = function ($data) {
+        };
+        $this->onBeforeCreate = $identity;
+        $this->onAfterCreate = $noop;
+        $this->onBeforeUpdate = $identity;
+        $this->onAfterUpdate = $noop;
+        $this->onBeforeRemove = $identity;
+        $this->onAfterRemove = $noop;
+    }
+
+    public function supports($data, array $context): bool {
+        return $this->dataPersister->supports($data, $context);
+    }
+
+    public function onBeforeCreate(Closure $onBeforeCreate): self {
+        $this->onBeforeCreate = $onBeforeCreate;
+
+        return $this;
+    }
+
+    public function onAfterCreate(Closure $onAfterCreate): self {
+        $this->onAfterCreate = $onAfterCreate;
+
+        return $this;
+    }
+
+    public function onBeforeUpdate(Closure $onBeforeUpdate): self {
+        $this->onBeforeUpdate = $onBeforeUpdate;
+
+        return $this;
+    }
+
+    public function onAfterUpdate(Closure $onAfterUpdate): self {
+        $this->onAfterUpdate = $onAfterUpdate;
+
+        return $this;
+    }
+
+    public function onBeforeRemove(Closure $onBeforeRemove): self {
+        $this->onBeforeRemove = $onBeforeRemove;
+
+        return $this;
+    }
+
+    public function onAfterRemove(Closure $onAfterRemove): self {
+        $this->onAfterRemove = $onAfterRemove;
+
+        return $this;
+    }
+
+    public function onCustomAction(CustomActionListener $customActionListener): self {
+        $this->customActionListeners[] = $customActionListener;
+
+        return $this;
+    }
+
+    public function onPropertyChange(PropertyChangeListener $propertyChangeListener): self {
+        $this->propertyChangedListeners[] = $propertyChangeListener;
+
+        return $this;
+    }
+
+    public function persist($data, array $context = []): object {
+        if ('post' === ($context['collection_operation_name'] ?? null)) {
```

Comment:
Hier würde ich mir erhoffen, dass auch GraphQL abgedeckt ist, das ist aus meiner Sicht der grösste Vorteil an diesem System. Für ein Beispiel siehe https://api-platform.com/docs/core/data-persisters/#decorating-the-built-in-data-persisters